src/lib/progress-bar/progress-bar.ts

```diff
@@ -113,6 +150,31 @@ export class MatProgressBar extends _MatProgressBarMixinBase implements CanColor
       return {transform: `scaleX(${scale})`};
     }
   }
+
+  ngAfterViewInit() {
+    if (!this._isNoopAnimation) {
+      // Run outside angular so change detection didn't get triggered on every transition end
+      // instead only on the animation that we care about (primary value bar's transitionend)
+      this._ngZone.runOutsideAngular((() => {
+        this._animationEndSubscription =
+            fromEvent(this._primaryValueBar.nativeElement, 'transitionend')
+            .pipe(filter(((e: TransitionEvent) =>
+              e.target === this._primaryValueBar.nativeElement)))
+            .subscribe(_ => this._ngZone.run(() => this.emitAnimationEnd()));
+      }));
+    }
+  }
+
+  ngOnDestroy() {
+    this._animationEndSubscription.unsubscribe();
+  }
+
+  /** Callback function on animation end, emit animationEnd event on determinate and buffer mode. */
```

Comment:
Method comment should not describe when its used, only what it does.  Should instead be:
```
/** Emit an animationEnd event if in determinate or buffer mode. */
```

Reply:
done.